book.h

```diff
@@ -0,0 +1,48 @@
+#ifndef BOOK_H
+#define BOOK_H
+#include <string>
+#include <iostream>
+#include <conio.h>
+using namespace std;
+class Book {
+        private:
+            string ISBN, title, author, publisher, categories;
+
+        public:
+            void setISBN(string ISBN) { this->ISBN = ISBN; }
```

Comment:
Try to use dot instead arrow.
Maybe it would work like this: `this.ISBN = isbn;`

Reply:
The C++ syntax does not support 'this.' notation, instead, in C++, object members are accessed using 'this->'.